packages/headless/src/features/facets/facet-set/facet-set-controller-actions.ts

```diff
@@ -43,3 +46,29 @@ export const executeToggleFacetSelect = createAsyncThunk<
     dispatch(executeSearch(analyticsAction));
   }
 );
+
+/**
+ * Toggles the facet value, deselect other facet values, and then executes a search with the appropriate analytics tag.
+ * @param facetId (string) The unique identifier of the facet (e.g., `"1"`).
+ * @param selection (FacetValue) The target facet value.
+ */
+export const executeToggleFacetSingleSelect = createAsyncThunk<
+  void,
+  {
+    facetId: string;
+    selection: FacetValue;
+  },
+  AsyncThunkSearchOptions<FacetSection & ConfigurationSection>
+>(
+  'facet/executeToggleFacetSingleSelect',
+  ({facetId, selection}, {dispatch, extra: {validatePayload}}) => {
+    const analyticsAction = getAnalyticsActionForToggleFacetSelect(
+      facetId,
+      selection
+    );
+    validatePayload({facetId, selection}, definition);
```

Comment:
It's pretty much a copy paste from the above `executeToggleFacetSelect`.

Would be pretty easy to just extract a local common fonction directly in this module.